libs/cohere/langchain_cohere/chat_models.py

```diff
@@ -52,6 +53,40 @@
 from langchain_cohere.llms import BaseCohere
 
 
+def _get_tool_results(messages: List[BaseMessage]) -> List[Dict[str, Any]]:
+    """Get tool_results from messages."""
+    tool_results = []
+    for message in messages:
+        if isinstance(message, ToolMessage):
+            tool_message = message
+            previous_ai_msgs = [
+                message
+                for message in messages[:-1]
+                if isinstance(message, AIMessage) and message.tool_calls
+            ]
+            if previous_ai_msgs:
+                previous_ai_msg = previous_ai_msgs[-1]
+                tool_message_name: Optional[str] = None
+                if tool_message.name:
+                    tool_message_name = tool_message.name
+                else:
+                    tool_message_name = tool_message.additional_kwargs.get("name")
+                tool_results.extend(
+                    [
+                        {
+                            "call": ToolCall(
+                                name=lc_tool_call["name"],
+                                parameters=lc_tool_call["args"],
+                            ),
+                            "outputs": [{"answer": tool_message.content}],
+                        }
+                        for lc_tool_call in previous_ai_msg.tool_calls
+                        if lc_tool_call["name"] == tool_message_name
```

Comment:
note: currently [unit tests maintain](https://github.com/langchain-ai/langchain-cohere/blob/a47964c494fe225936f564bebcfd3b25d1e3902c/libs/cohere/tests/unit_tests/test_chat_models.py#L55-L63) that IDs for tool calls are equal, and equal to the generation_id. So here I join on name.

Reply:
> note: currently unit tests maintain that IDs for tool calls are equal, and equal to the generation_id.

that's correct

> join on name. an alternative is to generate tool call IDs.

generating tool call IDs would be more correct - I think when there's duplicate tool names and multiple calls it won't work well?. If it helps, we're planning on adding more ids server side :)

Reply:
updated this so that we generate IDs